src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,478 @@
+        private int _absoluteExpirationIndex = -1;
+
+        private readonly SqlServerCacheOptions _options;
+        private readonly ILogger _logger;
+        private readonly ISystemClock _systemClock;
+        private readonly SqlQueries _sqlQueries;
+        private DateTimeOffset _lastExpirationScan;
+
+        public SqlServerCache(IOptions<SqlServerCacheOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options.Options;
+            _systemClock = _options.SystemClock;
+            _sqlQueries = new SqlQueries(_options.TableName);
+            _logger = loggerFactory.CreateLogger<SqlServerCache>();
+        }
+
+        public void Connect()
+        {
+            // Try connecting to the database to test if its available and also
+            // get and cache the ordinal positions of the columns for better performance.
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetTableSchema, connection);
+                connection.Open();
+
+                var reader = command.ExecuteReader(CommandBehavior.SchemaOnly);
+
+                _cacheItemIdIndex = reader.GetOrdinal(CacheItemId);
+                _cacheItemValueIndex = reader.GetOrdinal(CacheItemValue);
+                _expiresAtTimeUTCIndex = reader.GetOrdinal(ExpiresAtTimeUTC);
+                _slidingExpirationInTicksIndex = reader.GetOrdinal(SlidingExpirationInTicks);
+                _absoluteExpirationIndex = reader.GetOrdinal(AbsoluteExpiration);
+            }
+        }
+
+        public async Task ConnectAsync()
+        {
+            // Try connecting to the database to test if its available and also
+            // get and cache the ordinal positions of the columns for better performance.
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetTableSchema, connection);
+                await connection.OpenAsync();
+
+                var reader = await command.ExecuteReaderAsync(CommandBehavior.SchemaOnly);
+
+                _cacheItemIdIndex = reader.GetOrdinal(CacheItemId);
+                _cacheItemValueIndex = reader.GetOrdinal(CacheItemValue);
+                _expiresAtTimeUTCIndex = reader.GetOrdinal(ExpiresAtTimeUTC);
+                _slidingExpirationInTicksIndex = reader.GetOrdinal(SlidingExpirationInTicks);
+                _absoluteExpirationIndex = reader.GetOrdinal(AbsoluteExpiration);
+            }
+        }
+
+        public byte[] Get([NotNull] string key)
+        {
+            byte[] cacheItemvalue = null;
+            var utcNow = _systemClock.UtcNow;
+
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
```

Comment:
Consider specifying the size to avoid cache fragmentation on the server (the size of the parameter is part of the query, we learned this in EF and LINQ to SQL the hard way, but also feel free to perf-test it to make sure that is still the case :smile:)